astroid/nodes/node_classes.py

```diff
@@ -1343,7 +1343,9 @@ def postinit(
     """
 
     # This is set by inference.py
-    def _infer_augassign(self, context=None):
+    def _infer_augassign(
+        self, context: InferenceContext | None = None
+    ) -> Generator[InferenceResult | util.BadBinaryOperationMessage, None, None]:
         raise NotImplementedError
```

Comment:
I'm wondering, IMO it would make sense to change these to type definitions similar to the once for `assigned_stmts` and `infer_lhs`. They are set anyway, so there is no point in assigning a dummy function for it in the first place. What do you think?

Reply:
Fine with me!

Reply:
> Fine with me!

Let's try it then!